added buttons to export campaigns, and restricted exporting to campaigns with published quests; Closes #1850

thefoxwithin opened the pull request:
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/
### What?
Add buttons that lead to the export confirmation page. Much like importing, exporting a campaign that has no published quests is not allowed. The way we got around this with importing is not showing campaigns without published quests to decks other than the library. With exporting there's an explanation when hovering over the button.

### Why?
To allow users to get to the export confirmation page in ways that are explicitly protected and don't force url manipulation.

https://github.com/bytedeck/bytedeck/issues/1850

### How?
If the user is allowed to export (i.e. they are the owner or they are staff and `allow_staff_export=True`) the button is shown. If the campaign has no published quests then the button is disabled with help text explaining why.

### Testing?

- Wrote a test `test_export_get__disables_button_if_no_quests` that:
  - Creates a campaign with no published quests.
  - Renders the export confirmation page.
  - Confirms the page shows the alert message.
- Manually verified that campaigns with published quests display an **enabled export button** and navigate correctly to the confirmation page.

### Screenshots (if front end is affected)
<img width="1297" height="650" alt="image" src="https://github.com/user-attachments/assets/f0bd3dc6-7233-4fa9-adcc-9f17098185a4" />

<img width="1322" height="617" alt="image" src="https://github.com/user-attachments/assets/a785b961-86a8-486e-b7ca-f753bb00d496" />

<img width="1117" height="826" alt="image" src="https://github.com/user-attachments/assets/2ba76432-e3b7-4b0f-a4eb-31f3d5a57169" />


### Anything Else?
### Review request
@tylerecouture


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added an Export action for campaigns in the detail view and campaign list (visible when export is allowed); disabled and accessible when no published quests.  

* **Bug Fixes / UX**
  * Confirmation screen shows a red alert: "You cannot export a campaign with no published quests." Share button now has an explicit DOM id for client-side use.  

* **Permissions**
  * New site-config permission check controlling who can export to the library.  

* **Tests**
  * Added tests for empty-campaign export UI and export permission logic.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tylerecouture commented on src/library/templates/library/confirm_export_campaign.html:
```diff
@@ -41,8 +41,17 @@
+
+      {% if category_quest_count == 0 %}
+        <p class="text-danger mt-2">
+          You cannot export a campaign with no published quests.
```
Can you put this in a well at the top of the page similar how we did with the other detail pages?

thefoxwithin replied:
Changed it to be the well at the top, but also removes the rest of the confirmation page just like when the campaign already exists on the library.

<img width="1102" height="227" alt="image" src="https://github.com/user-attachments/assets/0db232de-8038-446b-a99a-878861991d17" />

<img width="1113" height="216" alt="image" src="https://github.com/user-attachments/assets/22752e9f-8e05-4dba-a160-3468b99be31f" />

I did this because the red well looked awkward if it was just included with the rest of the page. Here's what I mean
<img width="1000" height="799" alt="image" src="https://github.com/user-attachments/assets/6f51459d-41f9-4ad0-a605-5c3fa5bf5cbc" />

tylerecouture commented on src/quest_manager/views.py:
```diff
@@ -89,8 +89,16 @@ def get_queryset(self):
 
+        site_config = SiteConfig.get()
+        current_schema = getattr(self.request.tenant, "schema_name", None)
+        can_export = (
```
instead of duplicating this can_export code (here and below), let's put this in a function on the SiteConfig model.  Then it can have a proper unit test too.

`can_export = site_config.can_user_export_to_library(self.request.user)`

thefoxwithin committed:
added buttons to export campaigns, and restricted exporting to campaigns with published quests

added test and id to button

Changed alert to be in a red well and only thing rendered on page

centralized can_export logic with can_user_export_to_library

added full_clean before saving config

updated tests to explicitly test for an anonymous user

added more context to docstring and consolidated logic for can_user_export_to_library

removed damaging line that was causing errors

thefoxwithin commented on src/quest_manager/templates/quest_manager/category_detail.html:
```diff
@@ -19,6 +19,22 @@
 
+            {% if can_export %}
+              <a class="btn btn-default" role="button"
+                {% if object.quest_count != 0 %}
+                  href="{% url 'library:export_category' object.import_id %}"
+                  title="Export this Campaign to the Library"
```
I thought about adding in something like this, but there's no way that you can get to the export campaign button without published quests in the campaign without either making your own view, or doing more complicated manipulation.

And so I thought about earlier conversations where things like that were rejected due to being unnecessary fail safes.
